Streetcode/Streetcode.XUnitTest/MediatRTests/StreetCode/Streetcode/GetStreetcodeByFilterHandlerTests.cs

```diff
@@ -0,0 +1,127 @@
+﻿using System.Linq.Expressions;
+using FluentAssertions;
+using Moq;
+using Streetcode.BLL.DTO.AdditionalContent.Filter;
+using Streetcode.BLL.MediatR.Streetcode.Streetcode.GetByFilter;
+using Streetcode.DAL.Entities.Media.Images;
+using Streetcode.DAL.Entities.Streetcode;
+using Streetcode.DAL.Entities.Streetcode.TextContent;
+using Streetcode.DAL.Entities.Timeline;
+using Streetcode.DAL.Repositories.Interfaces.Base;
+using Xunit;
+
+namespace Streetcode.XUnitTest.MediatRTests.StreetCode.Streetcode;
+
+public class GetStreetcodeByFilterHandlerTests
+{
+    private readonly Mock<IRepositoryWrapper> _repositoryWrapper;
+    private readonly GetStreetcodeByFilterHandler _handler;
+
+    public GetStreetcodeByFilterHandlerTests()
+    {
+        this._repositoryWrapper = new Mock<IRepositoryWrapper>();
+        this._handler = new GetStreetcodeByFilterHandler(this._repositoryWrapper.Object);
+    }
+
+    [Theory]
+    [InlineData("Test", "Test Streetcode")]
+    [InlineData("TextContent", "TextContent")]
+    [InlineData("FactTitle", "FactTitle")]
+    [InlineData("TimelineTitle", "TimelineTitle")]
+    [InlineData("ArtDescription", "ArtDescription")]
+    public async Task Handle_ReturnsResults_WhenSearchQueryMatches(string searchQuery, string expectedContent)
+    {
+        // Arrange
+        var query = new GetStreetcodeByFilterQuery(new StreetcodeFilterRequestDTO { SearchQuery = searchQuery });
+
+        this.SetupRepositoryMock(expectedContent);
+
+        // Act
+        var result = await this._handler.Handle(query, CancellationToken.None);
+
+        // Assert
+        result.IsSuccess.Should().BeTrue();
+        result.Value.Should().ContainSingle();
+        result.Value.First().Content.Should().Be(expectedContent);
```

Comment:
Here will be warning `S6608 Prefer indexing instead of "Enumerable" methods on types implementing "IList"`
Consider using `result.Value[0]` instead of `result.Value.First()` just to not make the warning

Reply:
Hmm, in my Rider IDE I don't see any warnings like this. But ok, I will change it.